### PR TITLE
GH-12458: Align the occ config with the convention

### LINKED ISSUE #12458: Align occ endpoint config with other libs
- create a `config` folder and move the default occ config in it.
- Create a model folder for a file `occ-checkout-endpoints.model.ts` with a type augmented  OccEndpoint interface
- Identify checkout specific endpoints attributes from core and move them to the type augmented  OccEndpoint interface.

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-12458

### CODE REVIEW

File: feature-libs/checkout/occ/model/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './occ-checkout-endpoints.model';
```
Comment (WeizhengSap) [inline feature-libs/checkout/occ/model/index.ts]: I think we need this:
```
// Imported for side effects (module augmentation)
import './occ-checkout-endpoints.model';
```
but in user lib, I see we just export it. I will ask Stan about it.

Comment (WeizhengSap) [reply feature-libs/checkout/occ/model/index.ts]: don't forget to add this index to `public_api`

Comment (plabadie) [reply feature-libs/checkout/occ/model/index.ts]: For now, I added just the import, this its how bit was in the cart lib.

File: feature-libs/checkout/occ/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Imported for side effects (module augmentation)
+import './default-occ-checkout-config';
```
Comment (WeizhengSap) [inline feature-libs/checkout/occ/config/index.ts]: import the endpoint model, not this one, right?

File: feature-libs/checkout/occ/model/index.ts (hunk 1 above)
Comment (WeizhengSap) [reply feature-libs/checkout/occ/model/index.ts]: you added import to `config` not `endpoint` :)

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**